sqlx-postgres/src/arguments.rs

```diff
@@ -145,6 +145,14 @@ impl<'q> Arguments<'q> for PgArguments {
         write!(writer, "${}", self.buffer.count)
     }
 
+    fn merge(&mut self, other: Self) {
+        self.types.extend(other.types);
+        self.buffer.extend_from_slice(&other.buffer);
+        self.buffer.count += other.buffer.count;
+        self.buffer.patches.extend(other.buffer.patches);
+        self.buffer.type_holes.extend(other.buffer.type_holes);
```

Comment:
This is not going to work as expected. These contain offsets into the encoded data which need to be adjusted.